### PR TITLE
fix(dpp): harden nested document-property position parsing

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented

Hardens parsing of `position` on **nested** document-schema properties. The recursive nested-property parser performed a `position`-based sort whose result was never used, and that sort could fail on certain malformed schema inputs reached during contract processing. This removes the dead sort so nested properties parse robustly.

Reported by [@DanielDerefaka](https://github.com/DanielDerefaka) — thanks for the careful report.

## What was done?

- Removed the unused `position` sort in `try_from_schema`/`insert_values_nested`. Its result was never consumed — nested properties are emitted in source-map order by the existing `properties.iter()` loop just below it — so removing it changes no accepted-contract behavior while eliminating the failure path. Added a code comment so it isn't reintroduced.
- Added regression tests covering malformed nested positions on both the full-validation and check-tx parse paths, plus a valid-position case confirming parsing behavior is unchanged.

## How Has This Been Tested?

- `cargo test -p dpp --features validation --lib try_from_schema` — all 49 tests pass, including the 3 new regression tests and the existing nested-property tests (behavior unchanged).
- `cargo clippy -p dpp --features validation` and `cargo fmt --all` — clean.

## Breaking Changes

None. The removed sort had no effect (its output was discarded), so accepted contracts parse identically; only a previously-failing edge case now parses cleanly.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved schema parsing robustness by removing a risky sorting step that could cause panics when encountering edge-case position values, preserving original property order.

* **Tests**
  * Added unit tests to ensure nested property position values are handled safely across validation modes and to prevent regressions.

### CODE REVIEW

**Comment by thepastaclaw** [comment]
✅ Review complete (commit efa1d738)

File: packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v1/mod.rs [hunk 1]
```diff
@@ -756,6 +756,103 @@ mod tests {
...
+        /// wider set of malformed nested positions reached the same panic. None may panic now.
+        #[test]
+        fn malformed_nested_positions_do_not_panic_check_tx() {
+            let _ = parse(schema_with_nested_position(Value::Float(0.0)), false);
+            let _ = parse(schema_with_nested_position(Value::I64(-1)), false);
+            let _ = parse(
+                schema_with_nested_position(Value::U128(u64::MAX as u128 + 1)),
+                false,
+            );
+        }
```
Comment (thepastaclaw) [inline packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v1/mod.rs]: **🟡 Suggestion: Malformed-position regression tests only assert non-panic, not the parse outcome**

`malformed_nested_positions_do_not_panic_check_tx` and `nested_float_position_does_not_panic_full_validation` discard results with `let _ = parse(...)`, so they only lock in the no-panic invariant. The current behavior on these inputs is `Err` (propagated from `get_integer::<u64>()?` or the meta-schema), but a future refactor that silently accepts `Value::Float(0.0)`, `Value::I64(-1)`, or `Value::U128(u64::MAX as u128 + 1)` — or returns a misleading error variant — would still pass. Mirror the style of `valid_nested_positions_still_parse` and assert `Err(_)` (ideally the specific `ProtocolError` / `DataContractError` variant) on each malformed case so the post-fix contract is pinned.

<sub>source: ['claude']</sub>

File: packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/mod.rs [hunk 2]
```diff
@@ -204,17 +204,12 @@ fn insert_values_nested(
...
-                            .get_integer(property_names::POSITION)
-                            .expect("expected a position");
-                        pos_1.cmp(&pos_2)
-                    });
+                    // Nested properties are emitted below in source-map order (the
+                    // `properties.iter()` loop). A previous `position`-based `sort_by` here was
+                    // dead code — its sorted result was never read — and it read `position` with
+                    // `.expect()`, which could panic on adversarial schema input during block
+                    // execution. Removed (ordering unchanged); do not reintroduce a panicking
+                    // `position` read here.
```
Comment (thepastaclaw) [inline packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/mod.rs:216]: **💬 Nitpick: Comment should state the consensus-observable ordering invariant**

The replacement comment correctly warns against reintroducing a panicking `position` read, but the load-bearing invariant — that the resulting `IndexMap` insertion order is consensus-observable and historical contracts were emitted in source-map order — is left implicit. A future reader could introduce a *correctly-implemented* `position` sort and inadvertently soft-fork existing contracts whose source order ≠ position order. Make this explicit, e.g. "do not re-sort nested properties: the `IndexMap` insertion order is consensus-observable and historical contracts were committed in source-map order."

<sub>source: ['claude']</sub>

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

Targeted hardening PR that removes a dead `sort_by` in `insert_values_nested` whose `.expect("expected a position")` could panic on adversarial nested-schema input. The sorted vector was never consumed (the loop iterates `properties.iter()` directly), so accepted-contract ordering is unchanged. Tests cover both validation paths plus a positive baseline. Two minor test-strengthening suggestions; no blocking issues.

🟡 1 suggestion(s) | 💬 1 nitpick(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v1/mod.rs`:
- [SUGGESTION] packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v1/mod.rs:834-841: Malformed-position regression tests only assert non-panic, not the parse outcome
  `malformed_nested_positions_do_not_panic_check_tx` and `nested_float_position_does_not_panic_full_validation` discard results with `let _ = parse(...)`, so they only lock in the no-panic invariant. The current behavior on these inputs is `Err` (propagated from `get_integer::<u64>()?` or the meta-schema), but a future refactor that silently accepts `Value::Float(0.0)`, `Value::I64(-1)`, or `Value::U128(u64::MAX as u128 + 1)` — or returns a misleading error variant — would still pass. Mirror the style of `valid_nested_positions_still_parse` and assert `Err(_)` (ideally the specific `ProtocolError` / `DataContractError` variant) on each malformed case so the post-fix contract is pinned.
```
</details>

File: packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v1/mod.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v1/mod.rs]: Done in b95343229f — the malformed-position tests now assert the actual parse outcome (verified empirically), not just non-panic: a zero-fraction float parses in both modes (it's a valid integer per the meta-schema); negative / >u64::MAX positions are admitted on the check_tx path (meta-schema skipped) but rejected with a clean `ConsensusError` under full validation.

File: packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/mod.rs (hunk 2 above)
Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/mod.rs:216]: Done in b95343229f — the comment now states the load-bearing invariant: nested-property `IndexMap` insertion order is consensus-observable (historical contracts were committed in source-map order), so reintroducing even a correctly-implemented `position` sort would soft-fork existing contracts.

**Review by thepastaclaw — APPROVED** [review]
## Code Review

Incremental review of b95343229f over 43d6c0161e. Both prior findings are FIXED on the current head: the malformed-position regression tests now pin Ok/Err parse outcomes via assert_matches! (v1/mod.rs:826-886), and the comment at mod.rs:207-216 now documents the consensus-observable source-map IndexMap insertion-order invariant and explicitly warns against any nested-property sort. The genuinely new diff (test outcome assertions + invariant comment) directly resolves both carried-forward findings; no new in-scope issues identified.

**Review by thepastaclaw — APPROVED** [review]
## Code Review

All six agents (Claude + Codex across general/security/rust-quality) converged on a clean review: the PR removes a dead `position`-based sort in `insert_values_nested` whose `.expect()` could panic deterministically on adversarial nested-property positions (floats, negatives, or values exceeding u64), eliminating a chain-halt vector while preserving source-map ordering already provided by `properties.iter()`. Test coverage is layered (DPP unit tests across both validation modes plus a drive-abci end-to-end regression through `process_raw_state_transitions`). Prior-finding revalidation list is empty as expected (the b9534322 review was clean), so there are no carried-forward findings, and no new in-scope issues were identified in the latest delta.
